core/src/main/java/com/datastax/oss/driver/internal/core/context/DefaultNettyOptions.java

```diff
@@ -132,8 +132,7 @@ public void afterChannelInitialized(Channel channel) {
             ioShutdownQuietPeriod, ioShutdownTimeout, ioShutdownUnit));
     DefaultPromise<Void> closeFuture = new DefaultPromise<>(GlobalEventExecutor.INSTANCE);
     combiner.finish(closeFuture);
-    // stop the timer as well
-    timer.stop();
+    closeFuture.addListener(f -> timer.stop());
```

Comment:
Not directly related to this ticket, but I think it's a good idea to close the timer as late as possible.